os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -397,8 +407,46 @@ static void up_dumpstate(void)
 static void _up_assert(int errorcode) noreturn_function;
 static void _up_assert(int errorcode)
 {
-	/* Are we in an interrupt handler or the idle task? */
+#ifdef CONFIG_FAULT_MGR
+	int ret;
+	mqd_t fault_queue;
+	pid_t pid;
+	uint32_t ksram_segment_end = (uint32_t)__ksram_segment_start__ + (uint32_t)__ksram_segment_size__;
+	uint32_t kflash_segment_end = (uint32_t)__kflash_segment_start__ + (uint32_t)__kflash_segment_size__;
+
+	/* Check if the PC lies within kernel context */
+	if ((assert_pc >= (uint32_t)__ksram_segment_start__ && assert_pc <= ksram_segment_end) || (assert_pc >= (uint32_t)__kflash_segment_start__ && assert_pc < kflash_segment_end)) {
+
+		/* Reboot the system if fault is in kernel context */
+
+#ifdef CONFIG_BOARD_ASSERT_AUTORESET
+		boardctl(BOARDIOC_RESET, EXIT_SUCCESS);
+#else
+		(void)irqsave();
+		for (;;) {
+#ifdef CONFIG_ARCH_LEDS
+			board_autoled_on(LED_PANIC);
+			up_mdelay(250);
+			board_autoled_off(LED_PANIC);
+			up_mdelay(250);
+#endif
+		}
+#endif
+	} else {
+		fault_queue = mq_open(FAULT_MGR_MQ, O_WRONLY, 0666, 0);
+		DEBUGASSERT(fault_queue != (mqd_t)ERROR);
+
+		pid = getpid();
 
+		/* Send a message to fault manager with pid of the faulty task */
+		ret = mq_send(fault_queue, (const char *)&pid, sizeof(pid), 0);
```

Comment:
please add mq_close(fault_queue); and remove DEBUGASSERT

Reply:
Added, thanks